client/lib/purchases/index.ts

```diff
@@ -850,6 +850,10 @@ export function purchaseType( purchase: Purchase ) {
 		return null;
 	}
 
+	if ( isTieredVolumeSpaceAddon( purchase ) ) {
+		return null;
+	}
```

Comment:
Another way of going about this is to not render the `meta` if it's strictly `agency-offer` (you can do that below)

<img width="259" alt="image" src="https://github.com/Automattic/wp-calypso/assets/26530524/43ff375f-7d22-46e6-8c40-8a8c6e7cdae9">

Above, we'd modify the condition so it becomes `purchase.meta && purchase.meta !== 'agency-offer'` in case other flows/offers want to render some meaningful meta values.